test/verify/check-metrics

```diff
@@ -1419,6 +1419,8 @@ class TestMultiCPU(testlib.MachineCase):
 
 @testlib.skipOstree("no PCP support")
 @testlib.skipImage("pcp not currently in testing", "debian-testing")
+# FIXME: Works with TEST_SHOW_BROWSER; not obvious what happens
+@testlib.skipBrowser("Headless chromium spontaneously logs out Grafana", "chromium")
```

Comment:
I honestly don't think there's a lot of value filing a bug against Chromium, but it would be nice to have a link there...

At the same time: it *does* (currently) work with headless chromium being driven by CDP.  We make a secondary CDP connection if we're collecting coverage data.. I wonder if we could use an extra connection here to work around this issue somehow?

Reply:
Only by running the *entire* test with CDP commands. Which we can technically do, it'd just be an awful lot of duplication.

This (plus the same thing with TestPages.testAllLanguages) is pretty much the main embarrassing thing about this PR, everything else works fairly nicely. But I already spent hours on this, and I feel it's not worth letting this bitrot. There's a fair bunch of follow-ups to do anyway (and I just added this)

Reply:
Ya, I wasn't proposing to use CDP for everything — that's going in completely the wrong direction.  I was more musing about if having a CDP connection merely open at the side is enough to get the browser into this "not headless" state, even if you're driving it via webdriver...

Reply:
I don't think this is a deliberate behaviour -- it seems much too random for that (especially in `TestPages.testAllLanguages`, it fails at a different language or page -- feels more like some counter is hitting the ceiling or so? But adding `--disable-pushstate-throttle` didn't help). There's no error/log message anywhere.

But I see what you mean. I tried this:
```diff
--- test/verify/check-metrics
+++ test/verify/check-metrics
@@ -1419,8 +1419,6 @@ class TestMultiCPU(testlib.MachineCase):
 
 @testlib.skipOstree("no PCP support")
 @testlib.skipImage("pcp not currently in testing", "debian-testing")
-# FIXME: Works with TEST_SHOW_BROWSER; not obvious what happens
-@testlib.skipBrowser("Headless chromium spontaneously logs out Grafana", "chromium")
 class TestGrafanaClient(testlib.MachineCase):
 
     provision = {
@@ -1479,6 +1477,7 @@ class TestGrafanaClient(testlib.MachineCase):
         # Log into Grafana (usually http://127.0.0.2:3002 if you do it interactively)
         bg = testlib.Browser(mg.forward['3000'], label=self.label() + "-" + mg.label, machine=self)
         try:
+            bg.set_user_agent("Cockpit Test 1.23")
             bg.open("/")
             bg.wait_in_text("body", "Welcome to Grafana")
             bg.set_input_text("input[name='user']", "admin")
--- test/verify/check-pages
+++ test/verify/check-pages
@@ -254,13 +254,12 @@ OnCalendar=daily
         b.wait_visible('#password-group')
         b.wait_text("#password-group > label", "Passwort")
 
-    # FIXME: Works with TEST_SHOW_BROWSER; not obvious what happens
-    @testlib.skipBrowser("Headless chromium spontaneously logs out", "chromium")
     def testAllLanguages(self):
         # Test that pages do not oops
         m = self.machine
         b = self.browser
 
+        b.set_user_agent("Cockpit Test 1.23")
         def line_sel(i):
             return '.terminal .xterm-accessibility-tree div:nth-child(%d)' % i
 
```

`set_user_agent()` is a CDP call, which will start a CDP connection (that's lazily initialized) and keep it open. No change with either test.

In all cases, the only hint that we get is

> warn: transport closed: disconnected

i.e. the shell's watchdog notices that the session went away. (But that's the effect of the spontaneous logout, not the cause)

Reply:
:shrug: